File: backend/src/main/java/com/parking/domain/model/Reservation.java
package com.parking.domain.model;

import java.time.LocalDate;
import java.time.LocalDateTime;
import lombok.Data;

@Data
public class Reservation {
    private Long id;
    private String userId;
    private String parkingSpotId;
    private LocalDate startDate;
    private LocalDate endDate;
    private ReservationStatus status;
    private LocalDateTime checkInTime;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    public Reservation() {}

    public Reservation(Long id, String userId, String parkingSpotId, LocalDate startDate, LocalDate endDate, ReservationStatus status, LocalDateTime checkInTime, LocalDateTime createdAt, LocalDateTime updatedAt) {
        this.id = id;
        this.userId = userId;
        this.parkingSpotId = parkingSpotId;
        this.startDate = startDate;
        this.endDate = endDate;
        this.status = status;
        this.checkInTime = checkInTime;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
    }
} 

File: frontend/types/reservation.ts
export interface ReservationRequest {
  parkingSpotId: string;
  startDate: string; 
  endDate: string;  
  requiresElectricity: boolean;
}

export interface Reservation {
  id: number;
  userId: number;
  parkingSpotId: string;
  startDate: string;
  endDate: string;
  status: ReservationStatus;
  checkInTime?: string;
  createdAt: string;
  updatedAt: string;
}

export type ReservationStatus = 'PENDING' | 'CONFIRMED' | 'CANCELLED' | 'COMPLETED' | 'NO_SHOW' | 'CHECKED_IN';

export interface ReservationResponse {
  id: number;
  parkingSpotId: string;
  startDate: string;
  endDate: string;
  status: string;
  checkInTime?: string;
}

export interface ParkingSpot {
  id: string;  // Format: "A01", "B02", etc.
  row: string; // A, B, C, D, E, F
  number: number; // 1-10
  hasCharger: boolean;
} 